src/main/java/de/tum/in/www1/artemis/service/connectors/hades/HadesCITriggerService.java

```diff
@@ -0,0 +1,173 @@
+            BuildScriptGenerationService buildScriptGenerationService) {
+        this.restTemplate = restTemplate;
+        this.programmingLanguageConfiguration = programmingLanguageConfiguration;
+        this.buildScriptGenerationService = buildScriptGenerationService;
+    }
+
+    @Override
+    public void triggerBuild(ProgrammingExerciseParticipation participation) throws ContinuousIntegrationException {
+        log.debug("Triggering build for participation {} in Hades", participation.getId());
+        try {
+            var job = createJob(participation);
+            postJob(job);
+        }
+        catch (JsonProcessingException e) {
+            log.error("Error creating the build job for participation {}", participation.getId(), e);
+        }
+    }
+
+    @Override
+    public void triggerBuild(ProgrammingExerciseParticipation participation, String commitHash) throws ContinuousIntegrationException {
+        log.warn("Triggering builds with a commit hash is not supported for Hades. Triggering build without commit hash.");
+        triggerBuild(participation);
+    }
+
+    @Override
+    public void triggerBuild(ProgrammingExerciseParticipation participation, String commitHash, RepositoryType triggeredByPushTo) throws ContinuousIntegrationException {
+        log.warn("Triggering with a test push is not supported for Hades. Triggering build without test push.");
+        triggerBuild(participation, commitHash);
+    }
+
+    private void postJob(HadesBuildJobDTO job) {
+        try {
+            HttpHeaders headers = new HttpHeaders();
+            headers.add(HttpHeaders.AUTHORIZATION, "Basic " + Base64.getEncoder().encodeToString(((hadesUsername + ":" + hadesPassword).getBytes())));
+            headers.add(HttpHeaders.CONTENT_TYPE, "application/json");
+            HttpEntity<HadesBuildJobDTO> request = new HttpEntity<>(job, headers);
+
+            var response = restTemplate.postForEntity(hadesServerUrl + "/build", request, JsonNode.class);
+            log.debug("Hades response: {}", response);
+        }
+        catch (Exception e) {
+            log.error("Error triggering the build for {}", job, e);
+        }
+    }
+
+    private HadesBuildJobDTO createJob(ProgrammingExerciseParticipation participation) throws JsonProcessingException {
+
+        // Build Job Metadata
+        var metadata = new HashMap<String, String>();
+        metadata.put("PLAN_KEY", participation.getBuildPlanId());
+
+        var steps = new ArrayList<HadesBuildStepDTO>();
+
+        // Create Clone Step
+        // TODO: We need a solution to clone with less critical credentials - This requires changes in the localvc implementation
+        var cloneMetadata = new HashMap<String, String>();
+        cloneMetadata.put("REPOSITORY_DIR", "/shared");
+        cloneMetadata.put("HADES_TEST_USERNAME", gitUsername);
+        cloneMetadata.put("HADES_TEST_PASSWORD", gitPassword);
```

Comment:
I am not that big of a fan that you pass the credentials of the internal admin user here. Even with LocalCI we want to avoid this and therefore added SSH cloning for build agents with ephemeral keys.

Reply:
Yes this will be solved with https://github.com/ls1intum/Artemis/pull/8929. We can then use the git User of the student and their access token.

Reply:
Yeah. I kinda thought that either the access token or the SSH PR would be merged by now 🙃 
So this should be part of a follow-up once the token PR is done.